src/trace/context.ts

```diff
@@ -238,6 +245,162 @@ export function readTraceFromSQSEvent(event: SQSEvent): TraceContext | undefined
   return;
 }
 
+export function readTraceFromSNSSQSEvent(event: SQSEvent): TraceContext | undefined {
+  if (event.Records && event.Records[0] && event.Records[0].body) {
+    try {
+      const parsedBody = JSON.parse(event.Records[0].body) as SNSMessage;
+      if (
+        parsedBody.MessageAttributes &&
+        parsedBody.MessageAttributes._datadog &&
+        parsedBody.MessageAttributes._datadog.Value
```

Comment:
Conditions like these can be simplified to: `parsedBody?.MessageAttributes?._datadog?.Value`